### PR TITLE
[Issue #6843] Replace use of python_image_resize with Pillow image.resize

### PR DESCRIPTION
**Pull request by binkiesbane**
[Original Issue #6801]

GeoNode imports a dependency called 'python_resize_image'. This dependency has been replaced with Pillow.

python_resize_image.resizeimage has been replaced with pillow.ImageOps.fit

resizeimage has been removed from the list of imports.

python_resize_image has been removed from pip requirements.txt.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [ x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [ x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [ x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [ ] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [ ] The issue connected to the PR must have Labels and Milestone assigned
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [ ] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

File: geonode/geoserver/helpers.py [hunk 1]
```diff
@@ -2076,9 +2076,7 @@ def _render_thumbnail(req_body, width=240, height=200):
         im.thumbnail(
             (_default_thumb_size['width'], _default_thumb_size['height']),
             resample=Image.ANTIALIAS)
-        cover = resizeimage.resize_cover(
-            im,
-            [_default_thumb_size['width'], _default_thumb_size['height']])
+        cover = im.resize((_default_thumb_size['width'], _default_thumb_size['height']))
```
Comment (mtnorthcott) [inline geonode/geoserver/helpers.py:2079]: `resize(...)` may not be the best choice here. Check the [documentation on what `resizeimage.resize_cover(...)` actually does](https://github.com/VingtCinq/python-resize-image#resize_coverimage-size-validatetrue-resampleimagelanczos) as it is not as simple as just resizing the image. Consider the [`ImageOps` module for PIL](https://pillow.readthedocs.io/en/stable/reference/ImageOps.html#PIL.ImageOps.fit) as an alternative.

File: geonode/base/models.py [hunk 2]
```diff
@@ -1490,9 +1489,7 @@ def save_thumbnail(self, filename, image):
                     im.thumbnail(
                         (_default_thumb_size['width'], _default_thumb_size['height']),
                         resample=Image.ANTIALIAS)
-                    cover = resizeimage.resize_cover(
-                        im,
-                        [_default_thumb_size['width'], _default_thumb_size['height']])
+                    cover = im.resize((_default_thumb_size['width'], _default_thumb_size['height']))
```
Comment (mtnorthcott) [inline geonode/base/models.py:1492]: Same here.

**Review by mtnorthcott — CHANGES_REQUESTED** [review]

**Comment by binkiesbane** [comment]
Actioned change requests, replaced image.resize with ImageOps.fit which both crops and resizes thumbnails

**Review by mtnorthcott — DISMISSED** [review]

File: requirements.txt [hunk 3]
```diff
@@ -124,7 +124,6 @@ python-dateutil==2.8.1
 pytz==2020.5
 requests==2.25.1
 timeout-decorator==0.5.0
-python_resize_image==1.1.19
```
Comment (afabiani) [inline requirements.txt:127]: Can we remove the dependency from "setup.cfg" too?

**Review by afabiani — CHANGES_REQUESTED** [review]
Remove unused dependency from setup.cfg too

**Review by afabiani — APPROVED** [review]

**Pull request merged**